src/main/java/kotlinx/reflect/lite/KTypeProjection.kt

```diff
@@ -14,7 +14,7 @@ package kotlinx.reflect.lite
  * for more information.
  */
 @SinceKotlin("1.1")
-public data class KTypeProjection(
+public class KTypeProjection(
```

Comment:
Not sure why it was a data class, but it's probably not a good idea to have it in public API.

I also wonder if it should be an interface like all the other public declarations in this package (i.e `KType`, `KTypeParameter`, etc), with implementation encapsulated in `KTypeProjectionImpl`, or am I missing something?

Reply:
Long story short, we replicate the structure of `kotlin-reflect` here, so it's possible to drop-replace it with `reflect-lite` and tweaking **only** package names.

Reflect library was never properly designed, so it has some weird gotchas like this one indeed -- it definitely shouldn't be a data class. I suggest reverting it for the sake of behavioural compatibility for now, but we'll probably revise this API at some point